packages/editor/src/components/browser/DataTypeBrowser.tsx

```diff
@@ -1,13 +1,225 @@
-import { useState } from 'react';
-import { Input } from '../widgets';
+import { useMemo, useState, useEffect } from 'react';
+import { Checkbox, Collapsible, ExpandableCell, MessageText, SelectRow, Table, TableCell } from '../widgets';
 import { UseBrowserImplReturnValue } from './useBrowser';
+import {
+  ColumnDef,
+  ExpandedState,
+  RowSelectionState,
+  flexRender,
+  getCoreRowModel,
+  getExpandedRowModel,
+  getFilteredRowModel,
+  useReactTable
+} from '@tanstack/react-table';
+import { useEditorContext, useMeta } from '../../context';
+import { DataClass, JavaType } from '@axonivy/inscription-protocol';
 export const DATATYPE_BROWSER_ID = 'datatype' as const;
 
 export const useDataTypeBrowser = (): UseBrowserImplReturnValue => {
   const [value, setValue] = useState('datatype');
-  return { id: DATATYPE_BROWSER_ID, name: 'Datatype', content: <DataTypeBrowser value={value} onChange={setValue} />, accept: () => value };
+  return {
+    id: DATATYPE_BROWSER_ID,
+    name: 'Datatype',
+    content: <DataTypeBrowser value={value} onChange={setValue} />,
+    accept: () => value
+  };
 };
 
 const DataTypeBrowser = (props: { value: string; onChange: (value: string) => void }) => {
-  return <Input {...props} />;
+  const { context } = useEditorContext();
+
+  const [datatypeList, setDatatypeList] = useState<JavaType[]>([]);
+  const [mainFilter, setMainFilter] = useState<string>('');
+  const allDatatypes: JavaType[] = useMeta('meta/scripting/allTypes', { context, type: mainFilter }, []).data;
+  const dataClasses: DataClass[] = useMeta('meta/scripting/dataClasses', context, []).data;
+
+  const [typeAsList, setTypeAsList] = useState(false);
+  const [listtypeSelected, setListtypeSelected] = useState(false);
+
+  const [selectedDataType, setSelectedDataType] = useState<JavaType | undefined>();
+  const [showHelper, setShowHelper] = useState(false);
+
+  // Shorter useEffect for merging useMeta's (unfortunately i noticed that in the megat "allTypes" there are also types from the meta "dataClasses", so i created a new useEffect which removes these duplicates.)

```

Comment:
You're right, I've checked, and all types in 'dataClasses' are also present in 'allTypes'. Consequently, I could actually do away with this filter and just return 'allTypes' as soon as a search is initiated. However, the consequence of this is that if someone searches for a data class, the search is somewhat slower... I'm still looking into it with Reguel.

Reply:
I discussed it with Reguel. In theory, I could do without this filter (since everything is covered in 'allTypes'), but I would still prefer to keep this filter in the frontend. This speeds up the search and provides immediate feedback as you type your search query. I hope this is acceptable to you as well.